scripts/starknet/testing_utils.py

```diff
@@ -0,0 +1,158 @@
+        storage_table_file = os.path.join(
+            ARTIFACTS_DIR, f"storage_table_group_{group_count}.json"
+        )
+        storage_addr_count = 0
+        storage_table = {}
+        grouped_opcodes = []
+        l = len(op_group)
+        for opcode in op_group:
+            current_file = ""
+            if os.path.isdir(opcode):
+                continue
+            with open(opcode, "rt") as f:
+                grouped_opcodes += f.readlines()
+                if not grouped_opcodes[-1].endswith("\n"):
+                    grouped_opcodes[-1] += "\n"
+                current_file = ntpath.basename(f.name)
+            push_bytes = int(math.ceil(len(hex(storage_addr_count)[2:]) / 2))
+            grouped_opcodes += [
+                f"PUSH{push_bytes} {hex(storage_addr_count)}\n",
+                "SSTORE\n",
+            ]
+            # i doubt we will ever have more than 2^128 tests
+            low_gold, high_gold = get_gold_result(opcode)
+            storage_table[f"{current_file}"] = {
+                "storage_addr_low": storage_addr_count,
+                "gold_res_low": low_gold,
+                "gold_res_high": high_gold,
+            }
+            storage_addr_count += 1
+
+        with open(file_path, "w") as f:
+            f.writelines(grouped_opcodes)
+
+        with open(storage_table_file, "w") as f:
+            json.dump(storage_table, f, indent=4)
+        group_count += 1
+        storage_tables.append(storage_table)
+    return storage_tables
+
+
+def golden_test(result, gold_result, opcode):
+    assert (
+        int(result) == gold_result
+    ), f"""
+ ✗ {opcode} Test Failed
+ Expected: {gold_result}
+ Got:      {int(result)}"""
+    print(
+        f"""
+ ✓ {opcode}
+"""
+    )
+
+
+async def do_golden_test(opcode, test, abi, address):
+    args_low = get_storage_args(
+        abi,
+        address,
+        "get_storage_low",
+        [int(test["storage_addr_low"]), 0],
+    )
+    result_low = await invoke_or_call(args_low, call=True)
+    golden_test(result_low, int(test["gold_res_low"]), opcode)
+    args_high = get_storage_args(
+        abi,
+        address,
+        "get_storage_high",
+        [int(test["storage_addr_low"]), 0],
+    )
+    result_high = await invoke_or_call(args_high, call=True)
+    golden_test(result_high, int(test["gold_res_high"]), opcode)
+
+
+# insane hack, I know, but it makes the test 28x faster
+# python needs macros
+def generate_test_funcs(storage_tables, abis, addresses):
+    funcs = ""
+    for i in range(len(abis)):
+        funcs += "\n" + ",\n".join(
+            f'\tdo_golden_test("{opcode}", {test}, "{abis[i]}","{addresses[i]}")'
+            for opcode, test in storage_tables[i].items()
+        ) + ","
+    return funcs
+
+
+
+def get_compiled_contracts(contract_list):
+    a = []
+    for contract in contract_list:
+        a.append(f"{contract[:-6]}_compiled.json")
+    return a
+
+
+def get_abis(contract_list):
+    a = []
+    count = 1
```

Comment:
Seems to be unused

Reply:
It's being used

Reply:
I might be overlooking something, but this function is 5 lines long and I can't find any mention of `count` anywhere